tests/tools/calendar_test.py

```diff
@@ -115,3 +117,66 @@ async def test_calendar_tool_crashes_if_too_many_days(httpd: HTTPServer):
     assert False
   except ValueError as e:
     assert str(e) == "next_days must be at most 366"
+
+class MockHTTPRequestHandler(SimpleHTTPRequestHandler):
+    """Custom handler to serve different calendar files for testing."""
+    
+    def do_GET(self):
+        if "test-calendar-1.ics" in self.path:
+            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-1.ics")
+        elif "test-calendar-2.ics" in self.path:
+            ics_path = path.join(path.dirname(__file__), "..", "fixtures", "test-calendar-2.ics")
+        else:
+            self.send_error(404, "File not found")
+            return
+
+        with open(ics_path, 'rb') as f:
+            ics_content = f.read()
+
+        self.send_response(200)
+        self.send_header("Content-type", "text/calendar")
+        self.end_headers()
+        self.wfile.write(ics_content)
+
+
+def with_http_file_server(fn_async):
```

Comment:
Same motivation as above.

Aslo, we already have a `with_http_file_server` function defined above.

```suggestion
def with_multi_file_http_file_server(fn_async):
```